packages/driver/src/util/serialization/log.ts

```diff
@@ -0,0 +1,396 @@
+import _ from 'lodash'
+import { isSerializableInCurrentBrowser, preprocessForSerialization } from './index'
+import $dom from '../../dom'
+
+interface PreprocessedHTMLElement {
+  value?: any
+  type?: any
+  id?: string
+  tagName: string
+  attributes: { [key: string]: string }
+  innerHTML: string
+  serializationKey: 'dom'
+}
+
+interface PreprocessedFunction {
+  value: any
+  serializationKey: 'function'
+}
+
+/**
+ * Takes an HTMLElement that might be a <body> for a given snapshot or any other element, likely pertaining to log consoleProps,
+ * on the page that needs to be preprocessed for serialization. The method here is to do a very shallow serialization,
+ * by trying to make the HTML as stateful as possible before preprocessing.
+ *
+ * @param {HTMLElement} props - an HTMLElement
+ * @returns {PreprocessedHTMLElement} a preprocessed element that can be fed through postMessage() that can be reified in the primary.
+ */
+const preprocessDomElement = (props: HTMLElement) => {
+  // hydrate values in HTML copy so when serialized they show up correctly in snapshot. This is important for input boxes with typing and other 'value' attributes
+  props.querySelectorAll('input').forEach((input: HTMLInputElement) => {
+    input.setAttribute('value', input.value)
+    if (input.checked) {
+      input.setAttribute('checked', `${input.checked}`)
+    }
+  })
+
+  // TODO: figure out reifying option selection
+  // props.querySelectorAll('option').forEach((option) => {
+  //   if (option.selected) {
+  //     option.setAttribute('selected', option.selected)
+  //   }
+  // })
+
+  const el: PreprocessedHTMLElement = {
+    value: (props as HTMLInputElement)?.value,
+    type: (props as HTMLInputElement)?.type,
+    id: props?.id,
+    tagName: props.tagName,
+    attributes: {},
+    innerHTML: props.innerHTML,
+    serializationKey: 'dom',
+  }
+
+  // get all attributes and classes off the element
+  props.getAttributeNames().forEach((attributeName) => {
+    el.attributes[attributeName] = props.getAttribute(attributeName) || ''
+  })
+
+  return el
+}
+
+/**
+ * Takes an HTMLElement that might be a <body> for a given snapshot or any other element, likely pertaining to log consoleProps,
+ * on the page that needs to be preprocessed for serialization. The method here is to do a very shallow serialization,
+ * by trying to make the HTML as stateful as possible before preprocessing.
+ *
+ * @param {PreprocessedHTMLElement} props - a preprocessed element that was fed through postMessage() that need to be reified in the primary.
+ * @returns {HTMLElement} a reified element that can be used in a snapshot body or consoleProps.
+ */
```

Comment:
This comment block appears to be duplicated from above 😅

Reply:
Looks like I copy/pasta'ed for `reifyDomElement` 😂 . I'll update that function comment in a few to be more accurate!

Reply:
updated in [9269726](https://github.com/cypress-io/cypress/pull/20949/commits/926972626d9ce11493150637aba23573eb9d3e78)